pkgs/development/python-modules/slack-bolt/default.nix

```
@@ -0,0 +1,102 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, boddle
+, cherrypy
+, django
+, falcon
+, fastapi
+, flask
+, flask-sockets
+, moto
+, pyramid
+, pytest-asyncio
+, sanic
+, sanic-testing
+, slack-sdk
+, starlette
+, tornado
+, bottle
+, chalice
+, mock
+}:
+
+buildPythonPackage rec {
+  pname = "slack-bolt";
+  version = "1.16.1";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "slackapi";
+    repo = "bolt-python";
+    rev = "v${version}";
+    sha256 = "sha256-xzEsTx+44ae++WxArxO2rhCsot+ELuhc0aZ44MXY0c4=";
+  };
+
+  propagatedBuildInputs = [ slack-sdk ];
+
+  pythonImportsCheck = [
+    "slack_bolt"
+  ];
+
+  checkInputs = [
+    mock
+    pytestCheckHook
+    pytest-asyncio
+
+    # frameworks
+    bottle
+    boddle
+    cherrypy
+    django
+    falcon
+    fastapi
+    flask
+    flask-sockets
+    pyramid
+    tornado
+    starlette
+  ] ++ lib.optionals (!stdenv.isDarwin) [
+    # server types that are broken on darwin
+    chalice
+    moto
+    sanic
+    sanic-testing
+  ];
+
+  disabledTestPaths = [
+    # disable tests that require credentials
+    "tests/slack_bolt_async/oauth/test_async_oauth_flow.py"
+    "tests/slack_bolt/oauth/test_oauth_flow.py"
+  ] ++ lib.optionals stdenv.isDarwin [
+    # disable tests that test broken things on darwin
+    "tests/adapter_tests_async/test_async_sanic.py"
+    "tests/adapter_tests/aws/test_aws_chalice.py"
+    "tests/adapter_tests/aws/test_aws_lambda.py"
+    "tests/adapter_tests/aws/test_lambda_s3_oauth_flow.py"
+  ];
+
+  # disable tests that test auth or end to end integration
+  disabledTests = [
+    "test_events"
+    "test_interactions"
+    "test_lazy_listeners"
+    "test_oauth"
+  ];
+
+  # patch out pytest-runner
+  preBuild = ''
+    sed -i '/pytest-runner/d' ./setup.py
+  '';
+
+  meta = with lib; {
+    description = "A framework to build Slack apps using Python ";
```

More of an opinionated change, expect the whitespace which should be removed at the end.

```suggestion
    description = "A framework to build Slack apps";
```